src/modules/fw_pos_control_l1/FixedwingPositionControl.cpp

```diff
@@ -904,7 +891,24 @@ FixedwingPositionControl::control_position(const hrt_abstime &now, const Vector2
 				_tecs.set_height_error_time_constant(_param_fw_thrtc_sc.get() * _param_fw_t_h_error_tc.get());
 			}
 
-			tecs_update_pitch_throttle(now, alt_sp,
+		} else if (pos_sp_curr.type == position_setpoint_s::SETPOINT_TYPE_VELOCITY) {
```

Comment:
The position setpoint SETPOINT_TYPE_VELOCITY will be gone soon, can you entirely use `trajectory_setpoint`?

https://github.com/PX4/PX4-Autopilot/pull/17029

Reply:
If we remove SETPOINT_TYPE_VELOCITY then all logic in  control_position() depends on waypoints.
We can use yaw and yaw_valid in position_setpoint.msg to differentiate velocity control mode but must do many checks across the code in control_position() and logic will be not simple.

Reply:
@korotkoves The plan is to make everything depend on `trajectory_setpoint` and not waypoints